bot/exts/events/advent_of_code/views/dayandstarview.py

```diff
@@ -0,0 +1,67 @@
+from datetime import datetime
+
+import discord
+
+AOC_DAY_ANDSTAR_TEMPLATE = "{rank: >4} | {name:25.25} | {completion_time: >10}"
+
+
+class AoCDropdownView(discord.ui.View):
+    """Interactive view to filter AoC stats by Day and Star."""
+
+    def __init__(self, original_author: discord.Member, day_and_star_data: dict[str: dict], maximum_scorers: int):
+        super().__init__()
+        self.day = 1
+        self.star = 1
+        self.data = day_and_star_data
+        self.element_map = {element.custom_id: element for element in self.children}
+        self.maximum_scorers = maximum_scorers
+        self.original_author = original_author
+
+    def generate_output(self) -> str:
+        """Generates a formatted codeblock with AoC statistics based on the currently selected day and star."""
+        header = AOC_DAY_ANDSTAR_TEMPLATE.format(rank="Rank", name="Name",
+                                                 completion_time="Completion time (UTC)")
+        lines = [f"{header}\n{'-' * (len(header) + 2)}"]
+
+        for rank, scorer in enumerate(self.data[f"{self.day}-{self.star}"][:self.maximum_scorers]):
+            time_data = datetime.fromtimestamp(scorer['completion_time']).strftime("%I:%M:%S %p")
+            lines.append(AOC_DAY_ANDSTAR_TEMPLATE.format(datastamp="", rank=rank + 1, name=scorer['member_name'],
+                                                         completion_time=time_data))
+        joined_lines = "\n".join(lines)
+        return f"Statistics for Day: {self.day}, Star: {self.star}.\n ```\n{joined_lines}\n```"
+
+    async def interaction_check(self, interaction: discord.Interaction) -> bool:
```

Comment:
Please add an [`on_timeout`](https://discordpy.readthedocs.io/en/master/api.html#discord.ui.View.on_timeout) that will edit the message to have disabled components.

Reply:
I'm not sure how that could work. As far as I'm concerned the on_timeout callback won't receive any kind of interaction, hence even if I disable the elements, or just clear the View, it won't take an effect.
However, after the timeout the View cannot be used.

Reply:
Yeah that's exactly what I would've hoped to not get.

I understand why `on_timeout()` doesn't give us an interaction but it does complicate things. There's ways to work around it but I don't think it is worth the bother then.

Reply:
Reviving this conversation, see https://github.com/python-discord/sir-lancebot/pull/860#discussion_r716115195.

Reply:
Oh, right. It makes sense, I was _still thinking inside the box (view)_  😄 
Will push a patch, later today.